### PR TITLE
Nexus conda config

### PR DESCRIPTION
**Pull request by oliver7598**
# PR for issue
#1414 

## What is being addressed
DSVMs within Guacamole are currently deployed with anaconda although have no access to the anaconda repo for package management,

Currently conda install commands fail as they are trying to get the packages via https://repo.anaconda.com/ rather than a configured proxy.

## How is this addressed
- `repo.anaconda.com` added to `nexus_allowed_fqdns` variable to allow the external repo to be reached through the firewall
- `configure_nexus.sh` script amended to configure conda proxy endpoint
- Linux and Windows Guacamole user resources amended to configure conda proxy on User Resource deployment

### CODE REVIEW

**Review by marrobi — COMMENTED** [review]
@oliver7598 nice way to handle the requirement for conda binaries to be there.

There will likely be times where users need to configure conda/pip etc post deployment - for example in a container that hasn't had the cloud init scripts run.

Not necessarily part of this issue, but are the scripts available/can they be made available to be run interactively to configure the repositories?

**Comment by oliver7598** [comment]
@marrobi At current these script probably wont be available post deployment but it should be easy enough to extract them out and make them available.

In addition to this I'm assuming exposing the download link for Anaconda/PIP via Nexus would also be required to enable this?  If so I could pick this up after addressing another issue I am looking at and assuming no other higher priority issues arise.

**Comment by marrobi** [comment]
@oliver7598 oli
> In addition to this I'm assuming exposing the download link for Anaconda/PIP via Nexus would also be required to enable this?

To install the binaries? Good point, PIP can be done via apt which is mirrored, but might be need to make other binaries not covered by a package manager available via nexus in raw format.

File: scripts/configure_nexus.sh [hunk 1]
```diff
@@ -102,7 +102,7 @@ if [[ ${STATUS_CODE} == 404 ]]
 fi
 
 #Check if the repo already exists
-export STATUS_CODE=$(curl -iu admin:$NEXUS_PASS -X "GET" "${NEXUS_URL}/service/rest/v1/repositories/apt/proxy/pypi" -H "accept: application/json" -k -s -w "%{http_code}" -o /dev/null)
+export STATUS_CODE=$(curl -iu admin:$NEXUS_PASS -X "GET" "${NEXUS_URL}/service/rest/v1/repositories/proxy/pypi" -H "accept: application/json" -k -s -w "%{http_code}" -o /dev/null)
```
Comment (martinpeck) [inline scripts/configure_nexus.sh:105]: I'm probably missing something, but why remove the `apt` from the URL?

**Review by martinpeck — CHANGES_REQUESTED** [review]
Looks good, but I have one question

File: scripts/configure_nexus.sh (hunk 1 above)
Comment (oliver7598) [reply scripts/configure_nexus.sh:105]: Did this as a error correct as there is both a pypi_proxy_conf.json and apt-pypi_proxy_conf.json.
One is an apt repo and one is not. The curl request checks to see if the repo already exists before creating it so the standard proxy will never be created at the moment due to this error.

Comment (martinpeck) [reply scripts/configure_nexus.sh:105]: Ah, right. Got it.

**Review by martinpeck — APPROVED** [review]

**Comment by martinpeck** [comment]
Thanks @oliver7598 !

**Pull request merged**